### PR TITLE
AzureML/guidance Implementation

### PR DESCRIPTION
**Pull request by riedgar-ms**
Starting work on an AzureML implementation using `guidance`. So far, this just implements a simple zero-shot prompt.

### CODE REVIEW

**Comment by Harsha-Nori** [comment]
Could we promote the guidance related files outside of AzureML? I'd like them to be easily discoverable and runnable without the AzureML dependency. Wonder if the Azure files could point to a sibling "guidance" folder?

**Comment by riedgar-ms** [comment]
> Could we promote the guidance related files outside of AzureML? I'd like them to be easily discoverable and runnable without the AzureML dependency. Wonder if the Azure files could point to a sibling "guidance" folder?

It would just be an update in `constants.py`

**Comment by Harsha-Nori** [comment]
> > Could we promote the guidance related files outside of AzureML? I'd like them to be easily discoverable and runnable without the AzureML dependency. Wonder if the Azure files could point to a sibling "guidance" folder?
> 
> It would just be an update in `constants.py`

Ah nice, let's do that in the PR then and start building out a guidance/ folder with simple prompt implementations :).

**Review by Harsha-Nori — APPROVED** [review]
Seems reasonable, thanks!

File: azureml/components/jsonl_mmlu_fetch_component.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    type: string
+    optional: false
+    enum:
+    - anatomy
+    - astronomy
+    - clinical_knowledge
+    - college_biology
+    - college_medicine
+    - medical_genetics
+    - professional_medicine
```
[the remark this answers is not in the document]
Comment (riedgar-ms) [reply azureml/components/jsonl_mmlu_fetch_component.yaml:21]: I got tired of typing :-)

**Pull request merged**